File: src/main/java/net/softbell/bsh/iot/controller/rest/v1/IotAuthRestV1.java
package net.softbell.bsh.iot.controller.rest.v1;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import net.softbell.bsh.dto.rest.ResultDTO;
import net.softbell.bsh.iot.component.v1.IotAuthCompV1;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : IoT 인증 REST API 컨트롤러 V1
 */
@RestController
@RequestMapping("/api/rest/v1/iot/auth")
public class IotAuthRestV1
{
	@Autowired
	private IotAuthCompV1 iotAuthComp;
	
	@GetMapping("/token/check")
	public ResultDTO checkTokenAvailable(@RequestParam("token")String token)
	{
		// Field
		ResultDTO message;
		boolean isAvailable;
		
		// Check
		isAvailable = iotAuthComp.isTokenAvailable(token);
		
		// Message
		if (isAvailable)
			message = ResultDTO.builder().message("valid").build();
		else
			message = ResultDTO.builder().message("invalid").build();
		
		// Return
		return message;
	}
}

File: src/main/kotlin/net/softbell/bsh/iot/controller/rest/v1/IotAuthRestV1.kt
package net.softbell.bsh.iot.controller.rest.v1

import net.softbell.bsh.dto.response.ResultDto
import net.softbell.bsh.iot.component.v1.IotAuthCompV1
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/**
 * @author : Bell(bell@softbell.net)
 * @description : IoT 인증 REST API 컨트롤러 V1
 */
@RestController
@RequestMapping("/api/rest/v1/iot/auth")
class IotAuthRestV1 {
    @Autowired private lateinit var iotAuthComp: IotAuthCompV1

    @GetMapping("/token/check")
    fun checkTokenAvailable(@RequestParam("token") token: String): ResultDto {
        // Init
        val message = ResultDto()

        // Check
        val isAvailable = iotAuthComp.isTokenAvailable(token)

        // Message
        if (isAvailable)
            message.message = "valid"
        else
            message.message = "invalid"

        // Return
        return message
    }
}